integrations/java/iceberg-1.2/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseTableOperations.java

```diff
@@ -301,8 +301,6 @@ static Mono<GetTableResponseBody> handleCreateUpdateHttpError(
               casted, casted.getStatusCode().value() + " , " + casted.getResponseBodyAsString()));
     } else if (e instanceof WebClientResponseException) {
       return Mono.error(new WebClientResponseWithMessageException((WebClientResponseException) e));
-    } else if (e instanceof WebClientRequestException) {
```

Comment:
Let's double click on this as well while we are at it. 
WebClientResponseException

Reply:
Why is that not CSUE?

Reply:
We are converting 5xx errors from WebClientResponseException to CSUE above. This branch covers mainly other errors that could be 4xx.

Reply:
`NOT_IMPLEMENTED` this one should be safe to not categorized as CSUE, wdyt ?

Reply:
we have a branch above that is not needed due to catch all branch below..
seems like we can recognize non-CSUE when we get specific error code sent by the server which is safe, e.g. 4xx or not_implemented, all other scenarios should be CSUE because there is no response from server that guarantees failed commit

Reply:
> NOT_IMPLEMENTED this one should be safe to not categorized as CSUE

it is a 50x error, with ambassador in the middle, we cannot completely rule out if it might get surfaced in some way. I feel safer categorizing it a CSUE and running analysis on what gets dumped there.  

> we have a branch above that is not needed due to catch all branch below.

Thats true, I prefer we still keep the branch above. Its helping me with readability of expected CSUE and unexpected CSUE

Reply:
So we have covered all the cases 4xx and 5xx and for all unknown scenarios throwing CSUE although it is thrown for specific known cases and that should be fine to prevent data loss.